src/expr.rs

```diff
@@ -304,13 +304,24 @@ fn rewrite_if_else(context: &RewriteContext,
                                                     cond,
                                                     "let ",
                                                     " =",
-                                                    width - 3 - 2,
+                                                    try_opt!(width.checked_sub(3 + 2)),
                                                     offset + 3));
 
+    // Try to format if-else on single line.
+    if context.config.single_line_if_else {
+        if let result@Some(..) = single_line_if_else(context,
```

Comment:
I think this idiom is a bit odd, I'd prefer `let result = ...; if result.is_some() { return result; }`